packages/@storefront/core/src/services/tracker.ts

```diff
@@ -62,9 +63,21 @@ class TrackerService extends BaseService<TrackerService.Options> {
     }
   }
 
+  attachGbiEventMetadata = <S extends GbTracker.BaseEvent>(override: S): S => {
+    const { metadata = [] } = override;
+
+    if (metadata.some((item) => item.key === 'gbi')) {
+      return override;
+    } else {
+      // below needs to be cast as such in order to get around the restrictions of typeScript version 3.1.5, not allowing for generic types to be spread.
```

Comment:
- `TypeScript` with a capital `T`
- Prefix with `XXX:` if possible to indicate that this is a workaround
- Mention that the casts can be taken out with TypeScript 3.2.

Reply:
Done.